plugins/scale/scale-title-overlay.cpp

```diff
@@ -19,28 +47,90 @@ class view_title_overlay_t
     bool overflow = false;
 
     /**
-     * Get the transformed bounding box of the view, including the
-     * current transform
+     * Get the transformed bounding box of the view transformed by the given
+     * transformer, including the current transform, but not any padding.
      */
-    wlr_box get_transformed_bounding_box()
+    wlr_box get_transformed_bounding_box(wf::scale_transformer_t& tr)
     {
         wlr_box box = tr.get_transformed_view()->get_wm_geometry();
         box = tr.trasform_box_without_padding(box);
         return box;
     }
 
+    wlr_box get_transformed_bounding_box()
+    {
+        return get_transformed_bounding_box(tr);
+    }
+
+    /**
+     * Render the overlay text in our texture, cropping it to the size by
+     * the given box.
+     */
     void render_overlay(const wlr_box& box)
     {
         par.max_size = {box.width, box.height};
         auto view = tr.get_transformed_view();
-        auto res  = overlay.render_text(view->get_title(), par);
+        /* we always render the title of the topmost parent */
+        auto parent = find_toplevel_parent(view);
+        auto res    = overlay.render_text(parent->get_title(), par);
         overflow = res.width > overlay.tex.width;
         view->damage();
     }
 
     void render_overlay()
     {
-        render_overlay(get_transformed_bounding_box());
+        render_overlay((pos == position::CENTER) ? get_parent_box() :
+            get_transformed_bounding_box());
+    }
+
+    /**
+     * Get the bounding box of the topmost parent of this view without and
+     * padding added by scale's transformer.
+     */
+    wlr_box get_parent_box()
+    {
+        auto view = tr.get_transformed_view();
+        if (!view->parent)
+        {
+            return get_transformed_bounding_box();
+        }
+
+        view = find_toplevel_parent(view);
+
+        auto tmp =
+            view->get_transformer(wf::scale_transformer_t::transformer_name());
+        auto tr = dynamic_cast<wf::scale_transformer_t*>(tmp.get());
+        if (!tr)
+        {
```

Comment:
you could just assert(false), since this should never happen, if it does, at least we'll be able to detect such mistakes, otherwise, we might spent hours trying to figure out what is actually going on :)

Reply:
I'm not completely sure, since this is called from [here](https://github.com/dkondor/wayfire/blob/9f78ba1c08f9706ff0d04ed66db18f29e01893b1/plugins/scale/scale-title-overlay.cpp#L209) and I wonder if that line can be reached before the scale plugin added its transformer to a newly created dialog. Still, the test could be `if(!tmp)` and the `dynamic_cast` failing can be an error.

Reply:
Oh hm, this is a very good point. In this case, could you add a comment about this? i.e how we can reach that point.

Reply:
Ok, sure. I'm adding comments here. I'm not entirely sure if this will happen though, but I'm not that familiar with the sequence of events when views are created.